R/staggregate.R

```diff
@@ -228,11 +261,13 @@ staggregate_polynomial <- function(data, overlay_weights, daily_agg, time_agg =
   # Make each raster layer a data.table
   list_dt <- lapply(1:list_length, create_dt)
 
-  # Merge all data.tables together
+  # Merge all data.tables together if there are multiple
   clim_dt <- list_dt[[1]]
-  for(i in 2:list_length){
-    dt_m <- list_dt[[i]]
-    clim_dt <- merge(clim_dt, dt_m, by=c('x', 'y', 'date'))
+  if(list_length > 1 ){
+    for(i in 2:list_length){
+      dt_m <- list_dt[[i]]
+      clim_dt <- merge(clim_dt, dt_m, by=c('x', 'y', 'date'))
+    }
   }
```

Comment:
Is `create_dt` the exact same for all the transformations? (I think it used to be but not sure if it's changed a bit for each particular transformation). If the function is always the same though then maybe we can add it to the top of the script with the `as.data.table.raster()` function to avoid repeating it in every transformation function. If not, no worries it's totally fine as is.